Shopify/cli
# Introduce `shopify theme update init` and `shopify theme update check` commands

### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify/issues/420327 part (1/2).

### WHAT is this pull request doing?

#### New commands

This PR introduces 2 new commands:
- `shopify theme update init` that initializes an `update_extension.json` script
- `shopify theme update check` that checks if an `update_extension.json` script is valid

#### Schema validation

In order to power the `shopify theme update check` validation features, this PR relies on the `zod` schema validation library, which is currently used in `packages/cli-kit` and other CLI packages.

However, as discussed in the [thread](https://github.com/Shopify/cli/pull/1909#discussion_r1184700181) below, `zod` doesn't support loading a schema from a JSON schema file. Instead, `zod` requires its own domain-specific language (DSL) to declare schemas.

To use `zod` and benefit from its type inference, avoid adding a new library that duplicates `zod`'s functionality, and maintain a single schema declaration at [Shopify/theme-liquid-docs](https://github.com/Shopify/theme-liquid-docs), this PR introduces a new command in `packages/theme` called `pnpm run schema:generate`. This command automatically generates the `zod` schema based on the [Shopify/theme-liquid-docs](https://github.com/Shopify/theme-liquid-docs).

### How to test your changes?

- Run `shopify theme -h` and notice the new `shopify theme update` topic appears
- Run `shopify theme update -h` and notice the new commands appear
- Run `shopify theme update init` and notice it creates an `update_extension.json` script
- Run `shopify theme update check` and notice it checks if `update_extension.json` script is valid

<img width="40%" src="https://github.com/Shopify/cli/assets/1079279/a17ea826-ef45-4fd7-8d7e-e3ac455c67e7">

<img width="40%" src="https://github.com/Shopify/cli/assets/1079279/5adc0189-623f-4cbf-b59b-de1d8b139286">

<img width="40%" src="https://github.com/Shopify/cli/assets/1079279/9b9e7436-c6e0-4117-9525-5c82bf593931">

<img width="40%" src="https://github.com/Shopify/cli/assets/1079279/893b3719-d747-446b-9ef8-4d98ed72f1f8">


### Post-release steps

Update documentation.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
